### PR TITLE
Default Formatter implementation

### PR DESCRIPTION
**Pull request by goku321**
Added default implementation for `Formatter` interface.

Fixes: #69

### CODE REVIEW

File: expect.go [hunk 1]
```diff
@@ -237,6 +238,50 @@ func (DefaultRequestFactory) NewRequest(
...
+// added later.
+func (DefaultFormatter) EndAssertion(ctx Context) {}
+
+// Failure implements Formatter.Failure and reports failure.
+//
+// It formats the info from Context and Failure struct into
+// a string and passes it to Context.Reporter for reporting.
+func (DefaultFormatter) Failure(ctx Context, f Failure) {
+	errString := ""
+	if f.actual != nil {
```
Comment (gavv) [inline expect.go:268]: I think we should also add assertionName to Failure struct (e.g. "Equals", "Contains", etc) and print it here.

Comment (gavv) [reply expect.go:268]: Also, I suggest to rename Context.name to Context.TestName. And I guess we need to make all Context fields, except formatter, exported.

**Review by gavv — COMMENTED** [review]
Hi, sorry for delay! If you're still interested in this, here are a few comments.

**Comment by goku321** [comment]
Hi @gavv Thanks for your comments. Yes, I'm still interested but caught up with other tickets at the moment. Will surely return to this as soon as possible.

**Comment by Leryan** [comment]
@goku321 @gavv hi 👋 i’m very interested in these features, i could bring help if task sharing is possible :) i haven’t checked the current state of this PR in detail but i'm ready to spend several hours during the weekends. just ping me :)

**Comment by gavv** [comment]
@Leryan Thanks for your interest, the help is appreciated! You could help with finishing this PR, I hope @goku321 don't mind. After this is done, it would be possible to parallelize the work of switching various components to the Context, since those components can be changed independently.

File: expect.go (hunk 1 above)
Comment (Leryan) [reply expect.go:268]: the test (`testing.T`) reference could be inside the context?

**Comment by gavv** [comment]
Closing with respect to #150. See details in https://github.com/gavv/httpexpect/pull/98#issuecomment-1320941335

**Pull request closed without merging**